tests/set_option.c

```diff
@@ -0,0 +1,194 @@
+/* -*- Mode: C; indent-tabs-mode:nil -*- */
+/*
+ * Unit tests for libusb_set_option
+ * Copyright © 2023 Nathan Hjelm <hjelmn@cs.unm.edu>
+ * Copyright © 2023 Google, LLC. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "config.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include "libusbi.h"
+#include "libusb_testlib.h"
+
+#if defined(_WIN32) || defined(__CYGWIN__)
+/* Windows does not offer unsetenv. To simplify the test code just declare
+ * an empty implementation. */
+static void unsetenv(const char *env) {
```

Comment:
This fails on Cygwin: ../../tests/set_option.c:33:13: error: conflicting types for ‘unsetenv’
 static void unsetenv(const char *env) {
             ^~~~~~~~
In file included from /usr/include/stdlib.h:26:0,
                 from ../../tests/set_option.c:24:
/usr/include/cygwin/stdlib.h:27:5: note: previous declaration of ‘unsetenv’ was here
 int unsetenv (const char *);
     ^~~~~~~~

Reply:
And gives a warning both on MinGW and Cygwin:
../../tests/set_option.c:33:34: warning: unused parameter ‘env’ [-Wunused-parameter]
 static void unsetenv(const char *env) {
                                  ^~~

Reply:
Cygwin has `unsetenv`. So the code needs to be changed.

Reply:
Gotta love it. Let me work around this.